src/test/java/com/google/devtools/build/lib/bazel/rules/genrule/GenRuleWindowsTest.java

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.bazel.rules.genrule;
+
+import static com.google.common.truth.Truth.assertThat;
+import com.google.devtools.build.lib.actions.Artifact;
+import com.google.devtools.build.lib.analysis.actions.SpawnAction;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests of {@link BazelGenRule} on Windows.*/
+@RunWith(JUnit4.class)
+public class GenRuleWindowsTest extends BuildViewTestCase {
+
+  private static final Pattern BATCH_COMMAND_PATTERN =
+      Pattern.compile("(?<command>.*)");
+
+  private static final Pattern POWERSHELL_COMMAND_PATTERN =
+      Pattern.compile(".*'utf8';\\s+(?<command>.*)");
```

Comment:
I think we should assert that the script is written as utf-8. That's what we really want. Whether we tell PS this as a command line flag is irrelevant.

Reply:
No, if we don't set this and you do an `echo hello > foo` in powershell, `foo` will be encoded in UTF-16, that's probably not users want.

Reply:
Agreed! I think the test has the good intention but does the wrong assertion.

I meant to suggest you verify that the output file indeed is utf8-encoded. We know the way to do that is passing some flag to PS, but that's not the end result we want to test for.

To illustrate what I mean, if you wanted to integration-test the `zip` tool, you'd want to assert the contents of the produced `zip` file, not necessarily the command arguments that lead to generating that zip file. Does that make sense?

Reply:
Ah, I see. But this is not an integration test here. This is actually only a configured target test. We want to test the correct command is generated for the genrule, but the test won't actually execute the command. We have the integration test in `genrule_test.py`, in that test, we read the generated file with utf-8 encoding and compared the content is expected. Is that enough?

Reply:
And I'll rename the test to `GenruleWindowsConfiguredTargetTest`

Reply:
SGTM